pandas/core/generic.py

```diff
@@ -2146,8 +2147,10 @@ def tail(self, n=5):
         Returns last n rows
         """
         l = len(self)
-        if l == 0 or n == 0:
+        if l == 0:
             return self
+        if n == 0:
+            return self.iloc[l:]
```

Comment:
I think `self.iloc[0:0]` is clearer here. I don't think you need `l` anymore (was it ever needed?).

The side question here is: should we check that n>0? I don't think we need to... as this will make it kinda useful:  it'll return the rest of the dataframe except the head(n) == tail(len(df) - n).